gamelauncher_util.py

```diff
@@ -0,0 +1,73 @@
+import subprocess
+
+
+def get_steam_compat_lib():
+    """Find the library paths of a system.
+
+    Used for Steam's game drive
+    """
+    ldconfig_cmd = ["ldconfig", "-XNv"]
+    grep_cmd = ["grep", "^/.*:"]
+    paths = ""
+
+    try:
+        # TODO: Not ideal. Find a way to accomplish this simply in pure Python
+        result = subprocess.run(
+            ldconfig_cmd,
+            capture_output=True,
+            stderr=subprocess.DEVNULL,
+            text=True,
+            check=True,
+        )
+        result = subprocess.run(
+            grep_cmd, input=result.stdout, text=True, check=True, capture_output=True
+        )
+
+        for line in result.stdout.splitlines():
+            if not paths:
+                paths += line.split(":")[0]
+            else:
+                paths += ":" + line.split(":")[0]
+
+        return paths
+    except subprocess.CalledProcessError as err:
+        status = err.returncode
+        cmd = ldconfig_cmd + grep_cmd
+        chain = "".join(cmd)
+        raise subprocess.CalledProcessError(
+            f"An error occurred while executing {chain}: {status}"
+        )
+
+
+def get_steam_compat_install(envar):
+    """Find the mnt point of filesystem path.
+
+    Used for Steam's game drive
+    Expects envar to be STEAM_COMPAT_INSTALL_PATH
+    """
+    findmnt_cmd = ["findmnt", "-T", envar]
+    tail_cmd = ["tail", "-n", "1"]
+    awk_cmd = ["awk", "{ print $1 }"]
+
```

Comment:
The way `STEAM_COMPAT_LIBRARY_PATHS` are used by Steam quite easy. Steam has the concept of libraries - locations where Steam stores prefixes, games, shaders etc.... With ULWGL we don't really have that concept and each launcher (heroic,bottles,lutris) handles that differently.

The locations in `STEAM_COMPAT_LIBRARY_PATHS` are `:` separated paths, where at least one is a parent of `STEAM_COMPAT_INSTALL_PATH`. Game drive is only used for certain games, or when explicitly enabled via env var. Proton goes through the paths in `STEAM_COMPAT_LIBRARY_PATHS` and checks if `STEAM_COMPAT_INSTALL_PATH` is included in given `LIBRARY_PATH`. If the path like that was found it's being mounted as an `S:` drive to proton's prefix. If the `STEAM_COMPAT_LIBRARY_PATHS` is not defined Proton ignores the game drive request.

That variable is only used for game drive functionality. There are two things that game drive solves.
- make path shorter (for games that can't handle paths beyond certain length)
- allow launchers (e.g Genshin Impact, The Elder Scrolls Online) to properly calculate available size (in case of separate drive or location inaccessible from the container like flatpak)

By default we added `STEAM_COMPAT_INSTALL_PATH` there however it's not reliable as the fallback for that variable is parent of .exe. As that could lead to some issues we decided to find the nearest mount point that isn't `/`. That way we allow game drive to solve point two I showcased above. The first point may still cause minor issues if path is still beyond the character limit.